src/rest/RequestHandler.js

```diff
@@ -11,7 +11,7 @@ const Util = require('../util/Util');
 
 function parseResponse(res) {
   if (res.headers.get('content-type').startsWith('application/json')) return res.json();
-  return res.buffer();
+  return res.arrayBuffer();
```

Comment:
This returns an [`ArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer), if it's following the fetch API.

Fortunately, we do not have a method that returns `Promise<Buffer>` from a REST call, so this is not `semver: major` 😅

Reply:
https://github.com/nodejs/undici/blob/main/types/fetch.d.ts#L38 

types say that this returns a Buffer, but after testing this seems to be incorrect 😄

will submit a pr/issue over there to get this fixed

Reply:
If needed, I can convert to buffers with just
```js
Buffer.from(ArrayBuffer)
```

Reply:
I'm sure they were just referring to the fact that this is indeed different now, but since we only use it internally (and even there it can be either a buffer or array buffer) it doesn't matter.

Reply:
> but since we only use it internally (and even there it can be either a buffer or array buffer) it doesn't matter.

Actually, now that you mention it, it *does* matter:

https://github.com/discordjs/discord.js/blob/8cc3885739127ebc7023b6358ed0259ba7527250/src/managers/GuildMemberManager.js#L114-L116

This is broken after this PR given the current code, we need to either convert it to a buffer, or to use [`util.types.isArrayBuffer()`](https://nodejs.org/dist/latest-v16.x/docs/api/util.html#util_util_types_isarraybuffer_value).

Reply:
converted to Buffer in 6ac0b13

Reply:
I'm not sure whats more expensive here.

Just naively glancing over it, checking if its an array buffer instead of creating a new buffer should be lighter?

Reply:
We probably can catch all instances of those checks with just `instanceof Buffer` (or `Buffer.isBuffer`), but if I had to choose, I'd go for using `isArrayBuffer`.

~~It's also ironic how this PR is made with the premise of increasing the library's performance, while at the same time, also introducing overhead.~~

Bottom line is that it doesn't matter, because `Buffer` extends `Uint8Array`, which can take an `ArrayBuffer` as parameter, which is also used to efficiently share memory between typed arrays, so it's just a tiny wrapper on it, the memory isn't duplicated.

Of course, we don't read those buffers, so it's a bit useless to do the conversion.

Reply:
Okay, I removed the conversion from Buffer -> ArrayBuffer and checked if the data is an ArrayBuffer instead (other than the tests which were previously buffers so I kept those changes).

Reply:
I suggest using ArrayBuffer/Uint8Array instead of Buffer whenever possible, including Buffer to Web & Deno is not good

Reply:
discord.js and unidici are node-only

Reply:
I just mean that: working with Buffer is considered as a bad practices for cross compatibility support

Reply:
First, we intentionally removed browser support a while ago, see #5113.

Second, we have too many dependencies with Node.js to even be able to support Deno (or browser, for that matter).

And working with Buffer isn't a bad practice when the project is Node.js-only, it has a bunch of useful methods Uint8Array lacks of, and is a "Stability: 2" feature, which suggests it's both a stable and a recommended API.

Sure, we could change some stuff so we check for `Uint8Array` instances, but as far as Node.js goes, a lot of things take and return `Buffer` instances, and it can also confuse users when they have a `Buffer` but see a method that takes `Uint8Array`, as the names are very different (even though Buffer extends Uint8Array).

If anything, we can always look into cross-platform support with the TypeScript rewrite, but not for v13.